tests/ubench/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+include(ExternalProject)
+
+# List of micro benchmarks to build.
+
+set(bench_sources
+    accumulate_functor_values.cpp)
+
+# Set up google benchmark as an external project.
+
+set(gbench_src_dir "${CMAKE_CURRENT_SOURCE_DIR}/google-benchmark")
+set(gbench_install_dir "${PROJECT_BINARY_DIR}/gbench")
+
+set(gbench_cmake_args
+    "-DCMAKE_BUILD_TYPE=release"
+    "-DCMAKE_INSTALL_PREFIX=${gbench_install_dir}"
+    "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+    "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
+
+
+# Attempt to update git submodule if required.
+find_package(Git)
+if(NOT EXISTS "${gbench_src_dir}/.git")
+    if(GIT_FOUND)
+        exec_program("${GIT_EXECUTABLE}" "${CMAKE_CURRENT_SOURCE_DIR}"
+            ARGS submodule update --init google-benchmark)
+    else()
+        message(WARNING "google-benchmark submodule apparently not updated, but cannot find git")
```

Comment:
How about the following message: `Unable to update the google-benchmark submodule: git not found`

What happens if git is not found? From what I can tell the message is printed, and we continue trying to build the external project.

What happens if there is an error running the `git submodule update --init google-benchmark` command?